lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -114,11 +111,50 @@ class ReportUIFeatures {
     // There is only a sticky header when at least 2 categories are present.
     if (Object.keys(this.json.categories).length >= 2) {
       this._setupStickyHeaderElements();
-      this._document.addEventListener('scroll', this._updateStickyHeaderOnScroll);
+      const elToAddScrollListener = this._getScrollParent(this._dom.find('.lh-container', this._document));
+      elToAddScrollListener.addEventListener('scroll', this._updateStickyHeaderOnScroll);
       window.addEventListener('resize', this._updateStickyHeaderOnScroll);
     }
   }
 
+  /**
+   * Define a custom element for <templates> to be extracted from. For example:
+   *     this.setTemplateContext(new DOMParser().parseFromString(htmlStr, 'text/html'))
+   * @param {ParentNode} context
+   */
+  setTemplateContext(context) {
+    this._templateContext = context;
+  }
+
+  /**
+   * Finds the first scrollable ancestor of node. Falls back to the document.
+   * @param {HTMLElement} node
+   * @return {Node}
+   */
+  _getScrollParent(node) {
+    const isElement = node instanceof HTMLElement;
+    const overflowY = isElement && window.getComputedStyle(node).overflowY;
+    const isScrollable = overflowY !== 'visible' && overflowY !== 'hidden';
+
+    if (isScrollable && node.scrollHeight >= node.clientHeight) {
+      return node;
+    }
+
+    if (node.parentNode instanceof HTMLElement) {
+      return this._getScrollParent(node.parentNode);
+    } else {
+      return document;
+    }
+  }
```

Comment:
the HTMLElement checks are overkill if we're going to fall back to `document` in the end anyways (since it'll be the only non-element node encountered). I believe we can do something like
```js
/**
 * Finds the first scrollable ancestor of `element`. Falls back to the document.
 * @param {HTMLElement} element
 * @return {Node}
 */
_getScrollParent(element) {
  const {overflowY} = window.getComputedStyle(element);
  const isScrollable = overflowY !== 'visible' && overflowY !== 'hidden';

  if (isScrollable && element.scrollHeight >= element.clientHeight) {
    return element;
  }

  if (element.parentElement) {
    return this._getScrollParent(element.parentElement);
  }

  return document;
}
```

Does the `element.scrollHeight >= element.clientHeight` check always work out? What happens in DevTools with a short report (only one category, lots of passing)? Still works somehow? No scrolling so it doesn't matter anyways? Or should this return nothing in that case so we don't add a listener to whatever `document` is in devtools?

Reply:
ha, and this will help with the test errors :)

Reply:
> Does the element.scrollHeight >= element.clientHeight check always work out? What happens in DevTools with a short report (only one category, lots of passing)? 

oh, this would be handled by the `=` case. But now I'm wondering is it possible for `element.scrollHeight >= element.clientHeight` to ever be false :)